core/src/com/unciv/models/ruleset/unique/UniqueTriggerActivation.kt

```diff
@@ -78,7 +78,23 @@ object UniqueTriggerActivation {
                 return true
             }
             OneTimeFreeUnitRuins -> {
-                val unit = civInfo.getEquivalentUnit(unique.params[0])
+                var unit = civInfo.getEquivalentUnit(unique.params[0])
+                if ( unit.hasUnique(UniqueType.FoundCity) && civInfo.isOneCityChallenger()) {
+                    var found = false
+                    for (unitForReplace in ruleSet.units.values) {
+                        if(unitForReplace.getMatchingUniques(UniqueType.BuildImprovements)
+                           .any { it.params[0] == "Land" } ) {
+
+                            unit = civInfo.getEquivalentUnit(unitForReplace.name)
+                            found = true
+                            break
+                        }
+                    }
```

Comment:
can be simplified to replacementUnitl = ruleSet.units.values.firstOrNull{it.getMatchingUniques(UniqueType.BuildImprovements)
                           .any { it.params[0] == "Land" }} ?: return false